File: src/main/java/dev/tudubucket/whitelistedadmin/SendMessage.java
package dev.tudubucket.whitelistedadmin;

import me.clip.placeholderapi.PlaceholderAPI;
import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

public class SendMessage {

    public static void send(CommandSender sender, String message) {
        if (sender instanceof Player) {
            Player player = (Player) sender;
            String msg = PlaceholderAPI.setPlaceholders(player, message);
            player.sendMessage(ChatColor.translateAlternateColorCodes('&', msg));
        } else {
            String msg = PlaceholderAPI.setPlaceholders(null, message);
            Bukkit.getServer().getConsoleSender().sendMessage(ChatColor.translateAlternateColorCodes('&', msg));
        }
    }
}

File: src/main/kotlin/net/langdua/wladmin/SendMessage.kt
package net.langdua.wladmin

import me.clip.placeholderapi.PlaceholderAPI
import org.bukkit.Bukkit
import org.bukkit.ChatColor
import org.bukkit.command.CommandSender
import org.bukkit.command.ConsoleCommandSender
import org.bukkit.entity.Player

class SendMessage {
    companion object {
        fun send(sender: CommandSender?, message: String) {
            val msg =
                ChatColor.translateAlternateColorCodes(
                    '&',
                    PlaceholderAPI.setPlaceholders(
                        if (sender is Player && sender !is ConsoleCommandSender) sender else null,
                        message
                    )
                )
            if (sender != null) {
                sender.sendMessage(msg)
            } else {
                Bukkit.getServer().consoleSender.sendMessage(msg)
            }
        }
    }
}
